src/AtelierTomato.Markov.Service.Discord/DiscordObjectOIDBuilder.cs

```diff
@@ -5,15 +5,36 @@ namespace AtelierTomato.Markov.Service.Discord
 {
 	public class DiscordObjectOIDBuilder
 	{
-		public static async Task<DiscordObjectOID> Build(IGuild? guild, IChannel channel, ulong messageID, string instance = "discord.com")
+		public static async Task<DiscordObjectOID> BuildForMessage(IGuild? guild, IChannel channel, ulong messageID, string instance = "discord.com")
 		{
 			ulong serverID = guild?.Id ?? 0;
-			ulong categoryID, channelID, threadID;
+			var categoryToThread = await BuildCore(guild, channel);
+			return DiscordObjectOID.ForMessage(instance, serverID, categoryToThread.Category, categoryToThread.Channel, categoryToThread.Thread ?? 0, messageID);
+		}
+
+		public static async Task<DiscordObjectOID> BuildForThreadOrChannel(IGuild? guild, IChannel channel, string instance = "discord.com")
+		{
+			ulong server = guild?.Id ?? 0;
+			var categoryToThread = await BuildCore(guild, channel);
+			if (categoryToThread.Thread is not null)
+			{
+				return DiscordObjectOID.ForThread(instance, server, categoryToThread.Category, categoryToThread.Channel, categoryToThread.Thread.Value);
+			}
+			else
+			{
+				return DiscordObjectOID.ForChannel(instance, server, categoryToThread.Category, categoryToThread.Channel);
+			}
+		}
+
+		private static async Task<DiscordCategoryToThread> BuildCore(IGuild? guild, IChannel channel)

```

Comment:
not fond of how all of this is framed.
it would make more sense to build the full oid for the channel and then add the stuff that comes after that than to use some weird core distinction. in this sense, you also wouldn't need that random not very meaningful helper record you added.